data/test/scenarios/mapgen_filter.cfg

```diff
@@ -0,0 +1,103 @@
+#textdomain wesnoth-test
+
+# Check that the mapgen kernel's wesnoth.map.filter correctly handles "terrain" values.
+# This will load the generic_unit_test map, change all of the hexes that are between Alice's and
+# Bob's castles to void, and then check that the map looks as expected.
+{GENERIC_UNIT_TEST "mapgen_filter_terrain" (
+    map_generation=lua
+    map_data=""
+    map_file=""
```

Comment:
I think these are supposed to be omitted when making a generated map…

Reply:
Oops, I documented them on the Wiki and then forgot about it. Having them be empty strings is accepted by the C++, but as there's no way to unset an attribute in WML the only way to remove them would be to replace the use of GENERIC_UNIT_TEST. Is that worth doing?

Reply:
Ahhh, so they're here to null out the default from `GENERIC_UNIT_TEST`? As long as it doesn't cause any problems with the map generator, I think that should be fine. The only thing is, I brought it up because I thought I noticed someone recently saying it _does_ cause problems… although maybe that doesn't apply in unit tests?

Reply:
> I thought I noticed someone recently saying it *does* cause problems

Could you be thinking of my recent Wiki edit that says [setting it to a *non-empty* value causes implementation-defined behavior](https://wiki.wesnoth.org/index.php?title=MapGeneratorWML&curid=1446&diff=68460&oldid=68335)? Having a non-empty `map_data` skips map generation, and at this point any non-empty `map_file` will already have been read into `map_data`:

https://github.com/wesnoth/wesnoth/blob/ef0da66ec9be823f73490fd11434fae0edafdb00/src/saved_game.cpp#L498-L506

Reply:
That's probably what I was thinking of, which means there's no problem here at all.